dohq_artifactory/exception.py

```diff
@@ -1,2 +1,35 @@
+from json import JSONDecodeError
+
+import requests
+
+
 class ArtifactoryException(Exception):
     pass
+
+
+def raise_http_errors(response):
```

Comment:
Have no idea what the hell is going on in the function.

Could we simplify the function and make it more readable and add a few comments? You 100% have all this in mind, because you've been working with it a lot, but for one else who doesn't know all types of errors that requests can do - it's kinda tricky function

I'd rearrange it a little
```python
# Why do we need to rename method? Let's keep use the same name

def raise_for_status(response):
    try:
        response.raise_for_status()
    except requests.HTTPError as err:
        try:
            # we should use try on as few lines of code as possible
            error_list = err.response.json()
        except JSONDecodeError:
            raise ArtifactoryException(str(err))
      # Why it's error_list if it's actually dict??
       error_list = error_list.setdefault("errors", [{}])

       if isinstance(error_list[0], dict):
            # What type of message is it? 
            err_msg = error_list[0].setdefault("message", str(err))
        else:
            # What type of message is it? 
            err_msg = str(error_list[0])
   
      # Why it's err_msg if it contains fields or what is it? just  error_data 
       raise ArtifactoryException(err_msg) from err
```

Reply:
I have no objectives about renaming the method

it is an error List:
![image](https://user-images.githubusercontent.com/51964909/134889737-42190691-35d9-4ed4-bb80-102b5f8fdb13.png)

it is an err_msg, because it is a string, it is pure message

Reply:
![image](https://user-images.githubusercontent.com/51964909/134890558-db20cec3-90d7-4c59-b003-593da779be52.png)

Reply:
Let's document these tricks then in the code
```python
# lets rename err to exc also
except requests.HTTPError as exception:

try:
    error_list = exception.response.json()
excepct:
   # not a json response
   raise ArtifactoryException(...) from exception

# list of errors actually always have only one element, so we don't need to all of them
error_info = error_list[0]
if "message" not in error_info:
    # There's no detail message in the exception, raise an error with repr
    raise ArtifactoryException(str(exc)) from exception

# Artifactory sent a message about the error - we pass it as is
error_message = error_info['message"]
raise ArtifactoryException(err_msg) from exception
```

- We should avoid `setdefault` here, because it hides a lot of information about an operation and make code less readable.
- Also we should use `error_msg` and `exception`, because both `err_msg` and `err` look similar, but do not provide information about what they store

Reply:
I refactored the function, now should be clear at which place exception was raised and why that was raised by following comments in the code